src/Libraries/CoreNodes/Data.cs

```diff
@@ -51,16 +70,114 @@ private static object ToNative(JToken token)
             }
         }
 
+        /// <summary>
+        /// Parse implementation for converting JObject types to specific Dynamo objects (ie Geometry, Color, Images, etc) 
+        /// </summary>
+        /// <param name="jObject"></param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        private static object DynamoJObjectToNative(JObject jObject)
+        {
+            if (jObject.ContainsKey("$typeid"))
+            {
+                var typeid = jObject["$typeid"].ToString();
+
+                switch (typeid)
+                {
+                    //autodesk.math to abstract ProtoGeometry types
+                    case "autodesk.math:vector3d-1.0.0":
+                        return Vector.FromJson(jObject.ToString());
+
+                    case "autodesk.math:matrix44d-1.0.0":
+                        return CoordinateSystem.FromJson(jObject.ToString());
+
+                    //autodesk.geometry to abstract ProtoGeometry types
+                    case "autodesk.geometry:boundingbox3d-1.0.0":
+                        return BoundingBox.FromJson(jObject.ToString());
+
+                    case "dynamo.geometry:mesh-1.0.0":
+                        return Mesh.FromJson(jObject.ToString());
+
+                    //types supported by Goemetry.FromJson
+                    case "autodesk.math:point3d-1.0.0":
+                    case "dynamo.geometry:sab-1.0.0":
+                    case "dynamo.geometry:tsm-1.0.0":
+                    case "dynamo.geometry:rectangle-1.0.0":
+                    case "dynamo.geometry:cuboid-1.0.0":
+                    case "dynamo.geometry:solid-1.0.0":
+                    case string geoId when geoId.Contains("autodesk.geometry"):
+                        return Geometry.FromJson(jObject.ToString());
+
+                    //Dynamo types
+                    case "dynamo.graphics:color-1.0.0":
+                        return Color.ByARGB(
+                        (int)jObject["A"],
+                        (int)jObject["R"],
+                        (int)jObject["G"],
+                        (int)jObject["B"]);
+#if _WINDOWS
```

Comment:
Just curious, what is not supported in Linux here?

Reply:
System.Drawing  The Bitmap aspect.  You also find similar depreciation for the Image nodes.  I don't know if I need to guard the code here but most was trying to play it safe.  The `JsonConverter`  get marked with
```
 #if NET6_0_OR_GREATER
         [SupportedOSPlatform("windows")]
 #endif 
```